actors/evm/tests/asm.rs

```diff
@@ -0,0 +1,60 @@
+use etk_asm::ingest::Ingest;
+use evm::interpreter::opcode::OpCode::*;
+use fil_actor_evm as evm;
+
+#[allow(dead_code)]
+/// Creates a new EVM contract constructon bytecode (initcode), suitable for initializing the EVM actor.
+/// Arguments:
+/// - name is the name of the contract, for debug purposes.
+/// - init is the initializer code, which will run first at contract construction.
+/// - body is the actual contract code.
+pub fn new_contract(name: &str, init: &str, body: &str) -> Result<Vec<u8>, etk_asm::ingest::Error> {
+    // the contract code
+    let mut body_code = Vec::new();
+    let mut ingest_body = Ingest::new(&mut body_code);
+    ingest_body.ingest(name, body)?;
+    // the initialization code
+    let mut init_code = Vec::new();
+    let mut ingest_init = Ingest::new(&mut init_code);
+    ingest_init.ingest(name, init)?;
+    // synthesize contract constructor
+    let body_code_len = body_code.len();
+    let body_code_offset = init_code.len()
+        + 1 // PUSH4
+        + 4 // 4-bytes for code length
+        + 1 // DUP1
+        + 1 // PUSH4
+        + 4 // 4 bytes for the code offset itself
+        + 1 // PUSH1
+        + 1 // 0x00 -- destination memory offset
+        + 1 // CODECOPY
+        + 1 // PUSH1
+        + 1 // 0x00 -- source memory offset
+        + 1 // RETURN
+        ;
+    let mut constructor_code = Vec::<u8>::from([
```

Comment:
nit: we can just use an array with no vec.

Reply:
i want to append it later, so this looked more consistent.
I guess we could initialize the result vector with this array... dunno, doesnt look like a big diff.

having said all that, you are the bigclaw so if you think it matters we can change it :)

Reply:
Nah, doesn't matter. These are just nits.

(really, most of my comments are just nits)

Reply:
Although..., a better way to write this with a vec is:

```rust
vec![ ... ]
```

Otherwise, you'll write the intermediate array on the stack, then copy.

Reply:
ok, i can fix tmrw, i'll be adding some sugar for deduping code and some macros to simplify dispatch.